salt/states/pkg.py

```diff
@@ -2902,16 +2902,24 @@ def _uninstall(
         name, version, pkgs, normalize, ignore_epoch=ignore_epoch, **kwargs
     )
     if isinstance(targets, dict) and "result" in targets:
-        return targets
+        if action == "purge":
+            # found nothing, reset state return obj to empty list and check for removed to be purged
+            targets = []
+        else:
+            return targets
     elif not isinstance(targets, list):
-        return {
-            "name": name,
-            "changes": {},
-            "result": False,
-            "comment": "An error was encountered while checking targets: {}".format(
-                targets
-            ),
-        }
+        if action == "purge":
+            # found nothing, reset state return obj to empty list and check for removed to be purged
+            targets = []
+        else:
+            return {
+                "name": name,
+                "changes": {},
+                "result": False,
+                "comment": "An error was encountered while checking targets: {}".format(
+                    targets
+                ),
+            }
```

Comment:
**suggestion(non-blocking)**: could be:

```
if action != "purge":
    # indent the existing code here
else:
    targets = []
    old_removed = ...        
```

At least, I think that would logically the same.

Reply:
I was maintaining the same logic flow, and have to allow for this corner case of no installed packages to purge, but there are removed packages which should be purged.